functional-test/src/test/groovy/org/rundeck/tests/functional/api/job/JobExecutionSpec.groovy

```diff
@@ -1112,6 +1120,775 @@ class JobExecutionSpec extends BaseContainer {
+                        "handler executed successfully"
+                ), entries4)
+
+        when: "TEST: execution of job with keepgoing=true, errorhandler step succeeds (final wf executed)"
+        def stepOutfile5 = "/tmp/error-handler-out5.txt"
+        def job5Xml = generateErrorHandlerJob(
+                projectName,
+                "recover-handler3",
+                false,
+                true,
+                "echo handler executed successfully >> $stepOutfile5",
+                "echo final workflow step >> $stepOutfile5"
+        )
+        def created5 = JobUtils.createJob(projectName, job5Xml, client)
+        assert created5.successful
+
+        CreateJobResponse jobCreatedResponse5 = mapper.readValue(
+                created5.body().string(),
+                CreateJobResponse.class
+        )
+        def jobId5 = jobCreatedResponse5.succeeded[0]?.id
+
+        def runResponse5 = JobUtils.executeJob(jobId5, client)
+        // The job will execute, so the response will be successful
+        assert runResponse5.successful
+
+        Execution execId5 = mapper.readValue(runResponse5.body().string(), Execution.class)
+
+        // But the execution will fail
+        assert JobUtils.waitForExecutionToBe(
+                ExecutionStatus.SUCCEEDED.state,
+                execId5.id as String,
+                mapper,
+                client,
+                WaitingTime.LOW.milliSeconds,
+                WaitingTime.MODERATE.milliSeconds / 1000 as int
+        ).status == ExecutionStatus.SUCCEEDED.state
+
+        // So we read the output file
+        def execArgs5 = "cat $stepOutfile5"
+        def readResponse5 = client.doPostWithoutBody("/project/$projectName/run/command?exec=${execArgs5}")
+        def readResponseBody5 = readResponse5.body().string()
+        def parsedReadBody5 = mapper.readValue(readResponseBody5, RunCommand.class)
+        def readExecId5 = parsedReadBody5.execution.id
+
+        assert JobUtils.waitForExecutionToBe(
+                ExecutionStatus.SUCCEEDED.state,
+                readExecId5 as String,
+                mapper,
+                client,
+                WaitingTime.LOW.milliSeconds,
+                WaitingTime.MODERATE.milliSeconds / 1000 as int
+        ).status == ExecutionStatus.SUCCEEDED.state
+
+        def execOutputResponse5 = client.doGetAcceptAll("/execution/$readExecId5/output")
+        ExecutionOutput execOutput5 = mapper.readValue(execOutputResponse5.body().string(), ExecutionOutput.class)
+        def entries5 = execOutput5.entries.stream().map {it.log}.collect(Collectors.toList())
+
+        then: "test that errorhandler output was correct"
+        FileHelpers.assertLinesInsideEntries(
+                List.of(
+                        "handler executed successfully",
+                        "final workflow step"
+                ), entries5)
+
+        cleanup:
+        deleteProject(projectName)
+    }
+
+    def generateErrorHandlerJob(
```

Comment:
could you add doc describing this method, so the next person to use it knows exactly what error handler is going to be added